### PR TITLE
[FIX] Doxygen warnings in Range Module

### PR DESCRIPTION
**Pull request by joergi-w**
This PR improves the SeqAn documentation in the range module and removes all 134 warnings from the doxygen build.

### CODE REVIEW

File: include/seqan3/range/decorator/gap_decorator_anchor_set.hpp [hunk 1]
```diff
@@ -650,7 +659,10 @@ class gap_decorator_anchor_set
      * \param[in] rhs The right-hand side gap decorator to compare.
      * \returns A boolean flag indicating (in)equality of the aligned sequences.
      *
+     * \details
```
Comment (eseiler) [inline include/seqan3/range/decorator/gap_decorator_anchor_set.hpp]: I thought everything after a new line is implicitly details?

Comment (joergi-w) [reply include/seqan3/range/decorator/gap_decorator_anchor_set.hpp]: I thought so as well, but a look at the html output shows that the title line below is not parsed correctly if we omit the \details keyword.

**Comment by joergi-w** [comment]
As discussed I reverted the namespace nesting. This PR is ready for a final review.

File: include/seqan3/range/container/bitcompressed_vector.hpp [hunk 2]
```diff
@@ -971,33 +971,41 @@ class bitcompressed_vector
     }
     //!\}
 
-    //!\name Comparison operators
-    //!\{
+    /*!\name Comparison operators
+     * \{
```
[the remark this answers is not in the document]
Comment (joergi-w) [reply include/seqan3/range/container/bitcompressed_vector.hpp]: Doxygen does not parse this correctly and marks the following as undocumented. Apparently, we need to use block comments for subsequent lines that belong together.

File: include/seqan3/range/view/take.hpp [hunk 3]
```diff
@@ -208,17 +224,20 @@ class view_take : public ranges::view_interface<view_take<urng_t, exactly, or_th
             }
         }
 
+        //!\brief Checks whether `lhs` is equal to `rhs`.
         constexpr friend bool operator==(sentinel_type const & lhs, iterator_type const & rhs) noexcept(noexcept(rhs == lhs))
         {
             return rhs == lhs;
         }
 
+        //!\brief Checks whether `*this` is equal to `rhs`.
```
Comment (rrahn) [inline include/seqan3/range/view/take.hpp]:
```suggestion
        //!\brief Checks whether `*this` is not equal to `rhs`.
```

Comment (joergi-w) [reply include/seqan3/range/view/take.hpp]: Oops – thanks for spotting that!

**Pull request merged**